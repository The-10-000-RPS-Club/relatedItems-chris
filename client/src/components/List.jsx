/* eslint-disable react/prefer-stateless-function */
/* eslint-disable no-useless-constructor */
/* eslint-disable max-len */
/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */

import React from 'react';
import styled from 'styled-components';

const Wrap = styled.ul`
  display: flex;
  flex-direction: row;
  overflow: hidden;
  letter-spacing: 0;
  font-size: 1rem;
  line-height: 2rem;
  list-style: none;
  padding: 0px;
  height: 350px;
  scroll-behavior: smooth;
`;
const ListWrap = styled.section`
  height: 180px;
  width: 189px;
  padding: 7px 0px 0px 10px;
  
`;
const ImageSizer = styled.img`
  height: 120px;
  width: 160px;
  background-size: cover;
  background-repeat: no-repeat;
`;
const RedText = styled.span`
  color: #c83232;
  display: inline;
  font-family: Graphik,"Helvetica Neue",sans-serif;
    font-style: normal;
    font-weight: 300;
    letter-spacing: -.08px;
    font-size: .8rem;
    line-height: .75rem;
    margin-bottom: 4px;
`;
const ManufacturerText = styled.div`
  font-family: Graphik,"Helvetica Neue",sans-serif;
  font-style: normal;
  color: #292929;
  margin: 0px 0 0px;
  display: flex;
  align-items: center;
  font-weight: 340;
  letter-spacing: -.08px;
  font-size: .8rem;
  line-height: 1.1rem;
  left: 0;
  width: 100%;
  color: #292929;
`;
const ProductText = styled.span`
font-family: Graphik,"Helvetica Neue",sans-serif;
font-style: normal;
font-weight: 400;
letter-spacing: 0;
font-size: 1rem;
line-height: 1rem;
margin-bottom: 0;
`;
const NormalPrice = styled.div`
  text-decoration: line-through;
  display: inline;
  font-family: Graphik,"Helvetica Neue",sans-serif;
    font-style: normal;
    font-weight: 300;
    letter-spacing: -.08px;
    font-size: .8rem;
    line-height: .75rem;
    margin-bottom: 4px;
`;
const Price = styled.div`
  display: inline;
  font-family: Graphik,"Helvetica Neue",sans-serif;
    font-style: normal;
    font-weight: 300;
    letter-spacing: -.08px;
    font-size: .8rem;
    line-height: .75rem;
    margin-bottom: 4px;
`;
const StarCreator = styled.span`
color: #EDF0ED;
-webkit-text-stroke: .76px #BD7B2D;
height: 25px;

font-size: 17px;
position: relative;
float:left;

&:before {
  content: '★★★★★';
  opacity: .6;
}
&:after{
  font-size: 17px;
  content: "★★★★★";
  color: #F5CE6D;
  position: absolute;
  
  left: 0;
  top:0;
  width: ${(props) => props.rating * 15 || '0'}%;
  overflow: hidden;
}
`;
const NumberOfRatings = styled.span`
font-family: Graphik,"Helvetica Neue",sans-serif;
font-style: normal;
font-weight: 300;
letter-spacing: -.08px;
font-size: .8rem;
line-height: 1.9rem;
margin-bottom: 4px;
margin-right: 20px;
`;
const HoverUnderline = styled.div`
:hover {
  text-decoration: underline;
  cursor: pointer;
}
`;
const Section1 = styled.span`
display:flex;
a {
  postion: absolute;
  font-size: 3em;
  height: 45px;
  width: 45px;
  z-index: 1;
  margin: 130px 10px 0px 0px;
  &:nth-of-type(1){
    top: 0; bottom:0; left:0;
  }
}
`;
const Section2 = styled.span`
display:flex;

a {
  postion: absolute;
  color:#fff;
  text-decoration: none;
  height: 45px;
  width: 50px;
  text-align: center;
  z-index: 1;
  margin: 130px 0px 0px 0px;
  border: none;
  border-image: none;
}
`;
const ArrowShowLeft = styled.svg`
height: 20px;
width: 20px;
border-radius: 50%;
padding-top: 7px;
padding-left: 3px;
z-index: 9;
fill: grey;
`;
const ArrowShowRight = styled.svg`
height: 20px;
width: 30px;
border-radius: 50%;
padding-top: 7px;
padding-left: 3px;
z-index: 9;
fill: grey;

`;
const LeftButton = styled.button`
box-sizing: border-box;
border: none;
border-radius: 50%;
height: 40px;
bottom: initial;
display: flex;
width: 100%;
max-height: 40px;
max-width: 40px;
position: absolute;
z-index: 5;
cursor: pointer;
background-color: white;
margin: 120px 10px 0px 0px;
box-shadow: 0 0.4rem 0.4rem 0 rgba(12,11,8,.2);
transition: .20s linear;
outline-color: #b7e0fa;

&:hover {
  margin-top: 119px;
  box-shadow: 0 0.5rem 0.4rem 0.1rem rgba(12,11,8,.2);
  svg {
    fill: black;
  }
}
`;
const RightButton = styled.button`

border: none;
border-radius: 50%;
height: 40px;
bottom: initial;
display: flex;
width: 100%;
max-height: 40px;
max-width: 40px;
position: absolute;
z-index: 1;
cursor: pointer;
background-color: #f9f8f6;
box-shadow: 0 0.4rem 0.4rem 0 rgba(12,11,8,.2);
margin: 120px 0px 0px 1170px;
transition: .2s linear;
outline-color: #b7e0fa;


&:hover {
  margin: 118px 0px 0px 1170px;
  box-shadow: 0 0.5rem 0.4rem 0.1rem rgba(12,11,8,.2);
  svg {
    fill: black;
  }
}
#:active {
  border: 50%
}

`;

function List(props) {
  const { list } = props;
  const resultList = list.map((product) => {
    let salesPrice;
    if (product.onSale) {
      salesPrice = (
        <div id="related-item-price">
          <RedText>
            $
            {product.onSalePrice}
            {' '}
          </RedText>

          <NormalPrice>
            $
            {product.price}
            .99
          </NormalPrice>
        </div>
      );
    } else {
      salesPrice = (
        <div id="related-item-price">
          <Price>
            $
            {product.price}
            .99
          </Price>
        </div>
      );
    }
    return (
      <ListWrap key={product.product_id}>
        <li>

          <ImageSizer src={product.product_url} alt={product.product_id} />
          <HoverUnderline>

            <ManufacturerText>
              <div id="related-item-manufacturer">
                {' '}
                {product.manufacturer}
              </div>
            </ManufacturerText>

            <ProductText>
              <div id="related-item-product-name">
                {' '}
                {product.item_name}
              </div>
            </ProductText>

          </HoverUnderline>
          <div>
            <StarCreator rating={product.rating} />

            <NumberOfRatings>
              (
              {product.number_of_ratings}
              )
            </NumberOfRatings>
          </div>
          {salesPrice}
        </li>
      </ListWrap>
    );
  });

  return (
    <div>

      <a href="#section1">
        {' '}
        <LeftButton>
          <ArrowShowLeft aria-label="left scroll arrow">
            <path d="M 7.415 11 l 3.295 -3.295 a 1 1 0 0 0 -1.417 -1.412 l -4.98 4.98 a 0.997 0.997 0 0 0 -0.025 1.429 l 5.005 5.005 a 1 1 0 1 0 1.414 -1.414 L 7.414 13 H 19 a 1 1 0 0 0 0 -2 H 7.415 Z" />
          </ArrowShowLeft>
        </LeftButton>
        {' '}
      </a>

      <a href="#section2">
        {' '}
        <RightButton>
          <ArrowShowRight aria-label="right scroll arrow">
            <path d="M 16.585 13 l -3.295 3.295 a 1 1 0 0 0 1.417 1.412 l 4.98 -4.98 a 0.997 0.997 0 0 0 0.025 -1.429 l -5.005 -5.005 a 1 1 0 0 0 -1.414 1.414 L 16.586 11 H 5 a 1 1 0 0 0 0 2 h 11.585 Z" />
          </ArrowShowRight>
        </RightButton>
        {' '}
      </a>

      <Wrap>

        <Section1 id="section1">
          {resultList.slice(0, 6)}
        </Section1>
        <Section2 id="section2">
          {resultList.slice(6, resultList.Length)}
        </Section2>

      </Wrap>

    </div>

  );
}

// class List extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       leftDisplay: 'none',
//       rightDisplay: 'none',
//     };
//   }

//   render() {
//     const { list } = this.props;
//     const resultList = list.map((product) => {
//       let salesPrice;
//       if (product.onSale) {
//         salesPrice = (
//           <div id="related-item-price">
//             <RedText>
//               $
//               {product.onSalePrice}
//               {' '}
//             </RedText>

//             <NormalPrice>
//               $
//               {product.price}
//               .99
//             </NormalPrice>
//           </div>
//         );
//       } else {
//         salesPrice = (
//           <div id="related-item-price">
//             <Price>
//               $
//               {product.price}
//               .99
//             </Price>
//           </div>
//         );
//       }
//       return (
//         <ListWrap key={product.product_id}>
//           <li>

//             <ImageSizer image={product.product_url} />
//             <HoverUnderline>

//               <ManufacturerText>
//                 <div id="related-item-manufacturer">
//                   {' '}
//                   {product.manufacturer}
//                 </div>
//               </ManufacturerText>

//               <ProductText>
//                 <div id="related-item-product-name">
//                   {' '}
//                   {product.item_name}
//                 </div>
//               </ProductText>

//             </HoverUnderline>
//             <div>
//               <StarCreator rating={product.rating} />

//               <NumberOfRatings>
//                 (
//                 {product.number_of_ratings}
//                 )
//               </NumberOfRatings>
//             </div>
//             {salesPrice}
//           </li>
//         </ListWrap>
//       );
//     });

//     return (
//       <div>

//         <a href="#section1">
//           {' '}
//           <LeftButton>
//             <ArrowShowLeft>
//               <path d="M 7.415 11 l 3.295 -3.295 a 1 1 0 0 0 -1.417 -1.412 l -4.98 4.98 a 0.997 0.997 0 0 0 -0.025 1.429 l 5.005 5.005 a 1 1 0 1 0 1.414 -1.414 L 7.414 13 H 19 a 1 1 0 0 0 0 -2 H 7.415 Z" />
//             </ArrowShowLeft>
//           </LeftButton>
//           {' '}
//         </a>

//         <a href="#section2">
//           {' '}
//           <RightButton>
//             <ArrowShowRight>
//               <path d="M 16.585 13 l -3.295 3.295 a 1 1 0 0 0 1.417 1.412 l 4.98 -4.98 a 0.997 0.997 0 0 0 0.025 -1.429 l -5.005 -5.005 a 1 1 0 0 0 -1.414 1.414 L 16.586 11 H 5 a 1 1 0 0 0 0 2 h 11.585 Z" />
//             </ArrowShowRight>
//           </RightButton>
//           {' '}
//         </a>

//         <Wrap>
//           <Section1 id="section1">
//             {resultList.slice(0, 6)}
//           </Section1>
//           <Section2 id="section2">
//             {resultList.slice(6, resultList.Length)}
//           </Section2>

//         </Wrap>

//       </div>

//     );
//   }
// }

export default List;
