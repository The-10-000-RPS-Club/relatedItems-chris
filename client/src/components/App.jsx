/* eslint-disable import/extensions */
/* eslint-disable no-console */
import axios from 'axios';
/* eslint-disable no-unused-vars */
import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import List from './List.jsx';

const Wrapper = styled.div`
    font-size: 16px;
`;
const BorderBox = styled.div`
  max-width: 1200px;
  margin-right: auto;
  margin-left: auto;
  padding-left: 10px;
  padding-right: 10px;
  
`;
const Header = styled.h1`
  font-family: Graphik,Roboto,"Helvetica Neue",Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  letter-spacing: -.016rem;
  font-size: 1.3rem;
  line-height: 2.6rem;
  margin-left: 17%;
  
`;

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      productList: [],
    };
    this.getProductList = this.getProductList.bind(this);
  }

  componentDidMount() {
    this.getProductList();
  }

  getProductList() {
    axios({
      method: 'get',
      url: '/api/products/:id/relatedItems',
    })
      .then((data) => { this.setState({ productList: data.data }); })
      .catch((err) => err);
  }

  render() {
    const { productList } = this.state;
    let toRender = null;
    if (productList.length === 0) {
      toRender = <div>LOADING</div>;
    } else {
      toRender = <List list={productList} />;
    }
    return (
      <div>
        <div>
          <Header id="related-item-header"> Top picks for you </Header>
          <BorderBox>
            {toRender}
          </BorderBox>
        </div>
      </div>
    );
  }
}

export default App;
