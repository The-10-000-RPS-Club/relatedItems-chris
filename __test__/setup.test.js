/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */

import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import request from 'supertest';
import mongoose from 'mongoose';
import App from '../client/src/components/App';
import List from '../client/src/components/List';
import app from '../server/index';

Enzyme.configure({ adapter: new Adapter() });

// this is the data to test against

const productList = [{
  product_id: 100,
  manufacturer: 'Morar Group',
  item_name: 'Tasty Cotton Chips',
  rating: 0.753092278577937,
  number_of_ratings: 496,
  price: 267,
  onSalePrice: 225.84,
  product_url: 'https://fecimagesghrsea12.s3-us-west-1.amazonaws.com/pics/image100.jpg',
  onSale: true,
  __v: 0,
  _id: 12345,
},
{
  product_id: 90,
  manufacturer: 'Gusikowski, Kohler and Pacocha',
  item_name: 'Intelligent Plastic Shoes',
  rating: 3.655240113863547,
  number_of_ratings: 345,
  price: 925,
  onSalePrice: 785.84,
  product_url: 'https://fecimagesghrsea12.s3-us-west-1.amazonaws.com/pics/image90.jpg',
  onSale: true,
  __v: 0,
  _id: 1345,
},
{
  product_id: 89,
  manufacturer: 'Gleason, Pacocha and Hyatt',
  item_name: 'Fantastic Metal Towels',
  rating: 4.510381807800804,
  number_of_ratings: 0,
  price: 324,
  onSalePrice: 274.84,
  product_url: 'https://fecimagesghrsea12.s3-us-west-1.amazonaws.com/pics/image89.jpg',
  onSale: false,
  __v: 0,
  _id: 1235,
}];

describe('App component', () => {
  it('should render component', () => {
    expect(shallow(<App />).contains(<div id="list-container">Bar</div>)).toBe(false);
    expect(shallow(<App />).contains(<div>LOADING</div>)).toBe(true);
  });
});

describe('List component', () => {
  it('should contain rendered items', () => {
    const wrapper = shallow(<List list={productList} />);
    expect(wrapper.text()).toContain('Morar Group');
    expect(wrapper.text()).toContain('Tasty Cotton Chips');
    expect(wrapper.text()).toContain('267');
  });

  it('should render items', () => {
    const tree = renderer
      .create(<List list={productList} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

beforeAll((done) => {
  done();
});
afterAll((done) => {
  // Closing the DB connection allows Jest to exit successfully.
  mongoose.connection.close();
  done();
});

describe('Get /api/products/:id/relatedItems', () => {
  it('responds with json', (done) => {
    request(app)
      .get('/api/products/5/relatedItems')
      .set('Accept', 'application/json')
      .expect('Content-Type', /json/)
      .expect(200)
      .end((err) => {
        if (err) {
          return done(err);
        }
        return done();
      });
  });
});
