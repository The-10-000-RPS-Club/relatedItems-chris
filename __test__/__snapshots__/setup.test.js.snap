// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render items 1`] = `
<div>
  <a
    href="#section1"
  >
     
    <button
      className="sc-bqGHjH gIhGAv"
    >
      <svg
        className="sc-crzoUp hDghRy"
      >
        eslint-disable-next-line max-len
        <path
          d="M 7.415 11 l 3.295 -3.295 a 1 1 0 0 0 -1.417 -1.412 l -4.98 4.98 a 0.997 0.997 0 0 0 -0.025 1.429 l 5.005 5.005 a 1 1 0 1 0 1.414 -1.414 L 7.414 13 H 19 a 1 1 0 0 0 0 -2 H 7.415 Z"
        />
      </svg>
    </button>
     
  </a>
  <a
    href="#section2"
  >
     
    <button
      className="sc-ksluoS hhnNjD"
    >
      <svg
        className="sc-dIsAE bFBkRd"
      >
        <path
          d="M 16.585 13 l -3.295 3.295 a 1 1 0 0 0 1.417 1.412 l 4.98 -4.98 a 0.997 0.997 0 0 0 0.025 -1.429 l -5.005 -5.005 a 1 1 0 0 0 -1.414 1.414 L 16.586 11 H 5 a 1 1 0 0 0 0 2 h 11.585 Z"
        />
      </svg>
    </button>
     
  </a>
  <ul
    className="sc-bdnylx kLgSuC"
  >
    <span
      className="sc-kEqYlL gwnNYC"
      id="section1"
    >
      <section
        className="sc-gtssRu dQtEHJ"
      >
        <li>
          <img
            className="sc-dlnjPT jLmVSv"
          />
          <div
            className="sc-jrsJCI jsQOoP"
          >
            <div
              className="sc-eCApGN kjSuoB"
            >
              <div
                id="related-item-manufacturer"
              >
                 
                Morar Group
              </div>
            </div>
            <span
              className="sc-jSFkmK fxNbUe"
            >
              <div
                id="related-item-product-name"
              >
                 
                Tasty Cotton Chips
              </div>
            </span>
          </div>
          <div>
            <span
              className="sc-fujyUd emXjDs"
            />
            <span
              className="sc-pNWxx csLsBX"
            >
              (
              496
              )
            </span>
          </div>
          <div
            id="related-item-price"
          >
            <span
              className="sc-hKFyIo JPyHC"
            >
              $
              225.84
               
            </span>
            <div
              className="sc-gKAblj eFWMLi"
            >
              $
              267
              .99
            </div>
          </div>
        </li>
      </section>
      <section
        className="sc-gtssRu dQtEHJ"
      >
        <li>
          <img
            className="sc-dlnjPT hnDwtX"
          />
          <div
            className="sc-jrsJCI jsQOoP"
          >
            <div
              className="sc-eCApGN kjSuoB"
            >
              <div
                id="related-item-manufacturer"
              >
                 
                Gusikowski, Kohler and Pacocha
              </div>
            </div>
            <span
              className="sc-jSFkmK fxNbUe"
            >
              <div
                id="related-item-product-name"
              >
                 
                Intelligent Plastic Shoes
              </div>
            </span>
          </div>
          <div>
            <span
              className="sc-fujyUd EGWuD"
            />
            <span
              className="sc-pNWxx csLsBX"
            >
              (
              345
              )
            </span>
          </div>
          <div
            id="related-item-price"
          >
            <span
              className="sc-hKFyIo JPyHC"
            >
              $
              785.84
               
            </span>
            <div
              className="sc-gKAblj eFWMLi"
            >
              $
              925
              .99
            </div>
          </div>
        </li>
      </section>
      <section
        className="sc-gtssRu dQtEHJ"
      >
        <li>
          <img
            className="sc-dlnjPT cQfnuL"
          />
          <div
            className="sc-jrsJCI jsQOoP"
          >
            <div
              className="sc-eCApGN kjSuoB"
            >
              <div
                id="related-item-manufacturer"
              >
                 
                Gleason, Pacocha and Hyatt
              </div>
            </div>
            <span
              className="sc-jSFkmK fxNbUe"
            >
              <div
                id="related-item-product-name"
              >
                 
                Fantastic Metal Towels
              </div>
            </span>
          </div>
          <div>
            <span
              className="sc-fujyUd kQMwmh"
            />
            <span
              className="sc-pNWxx csLsBX"
            >
              (
              0
              )
            </span>
          </div>
          <div
            id="related-item-price"
          >
            <div
              className="sc-iCoHVE jnVIJE"
            >
              $
              324
              .99
            </div>
          </div>
        </li>
      </section>
    </span>
    <span
      className="sc-iqAbSa fVymTr"
      id="section2"
    />
  </ul>
</div>
`;
